src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2173,20 +2297,30 @@ private void onNetworkAbort(
         private void onNetworkWindow(
             WindowFW window)
         {
+            final long sequence = window.sequence();
+            final long acknowledge = window.acknowledge();
+            final int maximum = window.maximum();
             final long traceId = window.traceId();
             final long authorization = window.authorization();
             final long budgetId = window.budgetId();
-            int credit = window.credit();
             final int padding = window.padding();
 
+            assert acknowledge <= sequence;
+            assert sequence <= replySeq;
+            assert acknowledge >= replyAck;
+            assert maximum >= replyMax;
+
             state = AmqpState.openReply(state);
+            replyAck = acknowledge;
+            replyMax = maximum;
+            replyBudgetId = budgetId;
+            replyPad = padding;
 
-            replyBudget += credit;
-            replyPadding = padding;
+            assert replyAck <= replySeq;
 
             if (replyBudgetReserved > 0)
             {
-                final int reservedCredit = Math.min(credit, replyBudgetReserved);
+                final int reservedCredit = Math.min(replyMax, replyBudgetReserved);
```

Comment:
Changing `credit` to `replyMax` changes the intended logic for `reservedBudgetReserved`, agreed?